tensorboard/plugins/scalar/scalars_plugin.py

```diff
@@ -99,6 +151,22 @@ def scalars_impl(self, tag, run, output_format):
     else:
       return (values, 'application/json')
 
+  def _get_value(self, scalar_data_blob, dtype_enum):
+    """Obtains the value for a scalar event given the blob and dtype enum.
+
+    Args:
+      scalar_data_blob: The blob obtained from the database.
+      dtype_enum: The enum representing the dtype.
+
+    Returns:
+      The scalar value.
+    """
+    tensorflow_dtype = tf.DType(dtype_enum)
+    # Scalar tensors have no shape, so we index into them.
```

Comment:
Strictly speaking scalar tensors just have an empty shape.  I don't think we need the indexing of `[0]` here though - `np.asscalar()` should convert any size-1 tensor into a scalar regardless of shape.

```python
>>> np.asscalar(np.frombuffer(np.array([1.0])))
1.0
>>> np.asscalar(np.frombuffer(np.array(1.0)))
1.0
```

If we're concerned that we might have gotten a blob that's not actually a size-1 tensor despite filtering for scalar shapes (as described above) then I think it'd be better to instead pass `count=1` into `np.frombuffer()` so that we only parse out at most 1 item into the resulting array in the first place, rather than parsing more than one item needlessly and the discarding all but the first item.  However, since this is really a DB corruption issue I'm not sure it's worth trying to handle this gracefully by returning the 0th item of the array versus just failing.

Reply:
Ah, indeed, the call to `np.asscalar` does away with the shape! Removed the indexing.